src/Connectors/src/CloudFoundry/CloudFoundryServiceInfoCreator.cs

```diff
@@ -12,50 +13,48 @@ namespace Steeltoe.Connector.CloudFoundry
     public class CloudFoundryServiceInfoCreator : ServiceInfoCreator
     {
         private static readonly object _lock = new object();
-        private static IConfiguration _config;
         private static CloudFoundryServiceInfoCreator _me = null;
 
-        internal CloudFoundryServiceInfoCreator(IConfiguration config)
-            : base(config)
+        internal CloudFoundryServiceInfoCreator(IConfiguration configuration)
+            : base(configuration)
         {
-#pragma warning disable S3010 // Static fields should not be updated in constructors
-            _config = config;
-#pragma warning restore S3010 // Static fields should not be updated in constructors
             BuildServiceInfoFactories();
             BuildServiceInfos();
         }
 
-        public static new CloudFoundryServiceInfoCreator Instance(IConfiguration config)
+        public static new CloudFoundryServiceInfoCreator Instance(IConfiguration configuration)
         {
-            if (config == null)
+            if (configuration == null)
             {
-                throw new ArgumentNullException(nameof(config));
+                throw new ArgumentNullException(nameof(configuration));
             }
 
-            if (config == _config)
+            if (configuration == _me?.Configuration)
             {
                 return _me;
             }
 
             lock (_lock)
             {
-                if (config == _config)
+                if (configuration == _me?.Configuration)
```

Comment:
I suspect that you only expect to call this static method once.  Or, if more than once, with the same logical configuration.  If so, if the compared configs are not equal, should an error be thrown?  Or a warning logged?

Reply:
I know this is hit by the unit tests, its possibly it is here _only_ for testing purposes